repository.py

```diff
@@ -0,0 +1,552 @@
+#!/usr/bin/env python
+
+# Taken from https://opendev.org/openstack/sunbeam-charms/src/commit/5b37e0a6919668f23b8c7b148717714889fd4381/repository.py
+
+"""CLI tool to execute an action on any charm managed by this repository."""
+
+import argparse
+import glob
+import logging
+import os
+import pathlib
+import shutil
+import subprocess
+from dataclasses import dataclass
+
+import yaml
+
+ROOT_DIR = pathlib.Path(__file__).parent
+EXTERNAL_LIB_DIR = ROOT_DIR / "external" / "lib"
+BUILD_PATH = ROOT_DIR / "build"
+BUILD_FILE = "slurm-build.yaml"
```

Comment:
Tbh I don't think we need to worry about conflicting with other build systems here if we were to use the name `build.yaml`. Quick search seems to indicate the Dart is the only major language/ecosystem that uses a `build.yaml` file to control project builds. I'd have some questions if we had a Dart project in this repo 😶‍🌫️

If we don't want to have a generic `build.yaml` file, how about we use the name `charm-build.yaml` instead? I'm concerned that someone looking from the outside in at the Slurm charms would think that `slurm-build.yaml` dictates how we're building Slurm inside of the charms rather than dictating what charm libraries we're linking in as part of the charm packing process.

Reply:
changed to build.yaml